core/trino-main/src/test/java/io/trino/sql/planner/optimizations/TestRemoveUnsupportedDynamicFilters.java

```diff
@@ -477,7 +478,7 @@ private PlanNode removeUnsupportedDynamicFilters(PlanNode root)
         return getQueryRunner().inTransaction(session -> {
             // metadata.getCatalogHandle() registers the catalog for the transaction
             session.getCatalog().ifPresent(catalog -> metadata.getCatalogHandle(session, catalog));
-            PlanNode rewrittenPlan = new RemoveUnsupportedDynamicFilters(plannerContext).optimize(root, session, builder.getTypes(), new SymbolAllocator(), new PlanNodeIdAllocator(), WarningCollector.NOOP);
+            PlanNode rewrittenPlan = new RemoveUnsupportedDynamicFilters(plannerContext).optimize(root, session, builder.getTypes(), new SymbolAllocator(), new PlanNodeIdAllocator(), WarningCollector.NOOP, TableStatsProvider.EMPTY);
```

Comment:
`TableStatsProvider.EMPTY` is fine only because we know that `RemoveUnsupportedDynamicFilters` doesn't need stats.
an implementation that throws (`TableStatsProvider.UNIMPLEMENTED`) would be a better way of indicating that, but also constructing a real implementation (based on `metadata` that we have here) would be fine.